questions/services.py

```diff
@@ -620,8 +625,14 @@ def create_forecast(
     )
     forecast.save()
 
+    return forecast
+
+
+def after_forecast_actions(question: Question, user: User):
+    post = question.get_post()
```

Comment:
I think it's probably makes sense to move this block here (in the bottom of the `after_forecast_actions`):

```python
# Running forecast post triggers
    for post in posts:
        # There may be situations where async jobs from `create_forecast` complete after
        # `run_on_post_forecast` is triggered. To maintain the correct sequence of execution,
        # we need to ensure that `run_on_post_forecast` runs only after all forecasts have been processed.
        #
        # As a temporary solution, we introduce a 10-second delay before execution
        # to ensure all forecasts are processed.
        run_on_post_forecast.send_with_options(args=(post.id,), delay=10_000)
```

This will reduce code duplication